### PR TITLE
Extract AeInputAmount

### LINKED ISSUE #218: Remove flashing white border on "Amount" input on focusout.

### PR DESCRIPTION
**Pull request by CedrikNikita**
closes #218
and also fixes small issue with a different size of blocks and a small line between them:
Old                  vs                              New
image

### CODE REVIEW

File: src/components/Retip.vue [hunk 1]
```diff
@@ -250,20 +237,11 @@ export default {
...
-      }
-
-      &[type=number]:focus {
-        border-right: none;
-      }
-
-      &[type=number]:focus ~ .input-group-append .input-group-text {
-        border: .05rem solid $custom_links_color;
-        border-left: none;
+      .input-group /deep/ input.form-control {
```
Comment (venimus) [inline src/components/Retip.vue:243]: `/deep/` is some experimental selector that is deprecated, is this some boostrap junk you need to override?

File: src/components/Retip.vue [hunk 2]
```diff
@@ -250,20 +237,11 @@ export default {
     padding: 1rem;
     position: absolute;
 
-    .form-control {
-      color: $custom_links_color;
+    div {
```
Comment (venimus) [inline src/components/Retip.vue:240]:
```suggestion
    & > div {
```

File: src/components/Retip.vue (hunk 1 above)
Comment (davidyuk) [reply src/components/Retip.vue]: Actually, Vue has it's own `deep` that doesn't look like deprecated
https://vue-loader.vuejs.org/guide/scoped-css.html#deep-selectors

File: src/components/AeInputAmount.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
+<template>
+  <div>
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:2]: Can this empty wrapper be removed?

File: src/components/Retip.vue (hunk 1 above)
Comment (davidyuk) [reply src/components/Retip.vue]: As I know `deep` selectors are deprecated in native web components, in this case, `deep` shouldn't leak to client-side.

File: src/components/AeInputAmount.vue [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      border: 0;
+    }
+  }
+
+  .input-group-append > span.append__ae {
+    font-size: 0.75rem;
+    background: $buttons_background;
+    cursor: default;
+
+    & .ae {
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:66]:
```suggestion
    .ae {
```
`&` is useless if it is followed by space

File: src/components/Retip.vue [hunk 5]
```diff
@@ -57,31 +57,16 @@
...
-              :disabled="!isDataValid"
-              @click="retip()"
-            >
-              Retip
-            </ae-button>
-          </div>
+          <ae-input-amount
+            v-model.number="value"
+            :amount="value.toString()"
+          />
```
Comment (davidyuk) [inline src/components/Retip.vue:63]:
```suggestion
          <ae-input-amount v-model="value" />
```
why don't do like that?

File: src/components/AeInputAmount.vue [hunk 6]
```diff
@@ -0,0 +1,74 @@
+<template>
+  <div class="input-group">
+    <input
+      v-model="amount"
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:4]:
```suggestion
      :value="value"
```

File: src/components/AeInputAmount.vue [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import FiatValue from './FiatValue.vue';
+
+export default {
+  name: 'AeInputAmount',
+  components: {
+    FiatValue,
+  },
+  props: {
+    min: { type: Number, default: 0 },
+    step: { type: Number, default: 0.1 },
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:37]:
```suggestion
    step: { type: Number, default: 0.1 },
    value: { type: Number, required: true },
```

File: src/components/AeInputAmount.vue [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  data() {
+    return {
+      amount: 0,
+    };
+  },
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:43]:
```suggestion
```

File: src/components/AeInputAmount.vue [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+      aria-label="Default"
+      aria-describedby="inputGroup-sizing-mn"
+      @input="$emit('input', $event.target.value)"
+    >
+    <div class="input-group-append">
+      <span class="input-group-text append__ae">
+        <span class="ae">AE&nbsp;</span>
+        <fiat-value
+          :display-symbol="true"
+          :amount="amount.toString()"
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:20]:
```suggestion
          :amount="value.toString()"
```

File: src/components/Retip.vue [hunk 10]
```diff
@@ -57,31 +57,13 @@
...
-              </span>
-            </div>
-            <ae-button
-              :disabled="!isDataValid"
-              @click="retip()"
-            >
-              Retip
-            </ae-button>
-          </div>
+          <ae-input-amount v-model.number="value" />
```
Comment (davidyuk) [inline src/components/Retip.vue:60]:
```suggestion
          <ae-input-amount v-model="value" />
```

File: src/components/layout/SendTip.vue [hunk 11]
```diff
@@ -29,28 +29,8 @@
...
-                <span class="ae">AE&nbsp;</span>
-                <fiat-value
-                  :display-symbol="true"
-                  :amount="sendTipForm.amount.toString()"
-                />
-              </span>
-            </div>
-          </div>
+        <div class="col-md-4">
+          <ae-input-amount v-model.number="sendTipForm.amount" />
```
Comment (davidyuk) [inline src/components/layout/SendTip.vue:33]:
```suggestion
          <ae-input-amount v-model="sendTipForm.amount" />
```

File: src/components/AeInputAmount.vue [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+      class="form-control"
+      aria-label="Default"
+      aria-describedby="inputGroup-sizing-mn"
+      @input="$emit('input', $event.target.value)"
+    >
+    <div class="input-group-append">
+      <span class="input-group-text append__ae">
+        <span class="ae">AE&nbsp;</span>
+        <fiat-value
+          :display-symbol="true"
```
Comment (davidyuk) [inline src/components/AeInputAmount.vue:19]:
```suggestion
          display-symbol
```

File: src/components/Retip.vue (hunk 1 above)
Comment (venimus) [reply src/components/Retip.vue]: Good to know. However I was actually asking isn't it an overshoot to use so complex selector for such a simple case?

File: src/components/AeInputAmount.vue (hunk 7 above)
Comment (venimus) [reply src/components/AeInputAmount.vue:37]: Isn't it better to use `type: Number`? Also change the type of `amount` in the parent view (which is currently String)

**Comment by mradkov** [comment]
@CedrikNikita Please apply the changes requested and resolve the conflicts so we can merge.

**Comment by mradkov** [comment]
@stanislav-slavov please let me know when this is ready for review.

**Comment by stanislav-slavov** [comment]
> @stanislav-slavov please let me know when this is ready for review.

Please review it

**Comment by mradkov** [comment]
Checks do not pass.

**Pull request merged**